### PR TITLE
Test point calculation with cristin data from 2022

### PR DESCRIPTION
**Pull request by monaullah**
Test point calculation with cristin data from 2022:
Tested cases with institution collaboration and international collaboration with the four types: AcademicArticle, AcademicChapter, AcademicMonograph, AcademicLiteratureReview

### CODE REVIEW

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviNviCandidateHandlerTest.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package no.sikt.nva.nvi.evaluator;
```
Comment (truhacevkir) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviNviCandidateHandlerTest.java:2]: I guess filename should be EvaluateNviCandidateHandlerTest,

**Review by truhacevkir — APPROVED** [review]
Looks good, the only thing I would do is to refactor this:
```
var message = sqsClient.getSentMessages().get(0);
        var body = attempt(() -> objectMapper.readValue(message.messageBody(), 
CandidateEvaluatedMessage.class)).orElseThrow();
```
To something like this, since this two lines are repeated in every test:
`var body = getCandidateEvaluatedMessage();`

Also in test you are actually comparing our points against points calculated by Cristin from 2022. 
Maybe test should name something like shouldReturnSamePointsAsPointsCalculatedByCristin..... Or something that says that we are comparing against previously calcucated points.

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviCandidateTestWithCristinData.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    private FakeSqsClient sqsClient;
+    private ByteArrayOutputStream output;
+    private AuthorizedBackendUriRetriever uriRetriever;
+
+    @BeforeEach
+    void setUp() {
+        var s3Client = new FakeS3Client();
+        s3Driver = new S3Driver(s3Client, BUCKET_NAME);
+        sqsClient = new FakeSqsClient();
+        SqsMessageClient queueClient = new SqsMessageClient(sqsClient);
```
Comment (brinxmat) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviCandidateTestWithCristinData.java:74]: var

File: nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviCandidateTestWithCristinData.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        uriRetriever = mock(AuthorizedBackendUriRetriever.class);
+        var calculator = new CandidateCalculator(uriRetriever);
+        var storageReader = new FakeStorageReader(s3Client);
+        var evaluatorService = new EvaluatorService(storageReader, calculator);
+        handler = new EvaluateNviCandidateHandler(evaluatorService, queueClient);
+        output = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnSamePointsAsPointsCalculatedByCristinForAcademicArticleFrom2022() throws IOException {
```
Comment (brinxmat) [inline nvi-evaluator/src/test/java/no/sikt/nva/nvi/evaluator/EvaluateNviCandidateTestWithCristinData.java:87]: The tests are well laid out and clearly define what they are testing. Nice.

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Pull request merged**